### PR TITLE
8300659: Refactor TestMemoryAwareness to use WhiteBox api for host values

### PR DESCRIPTION
**Pull request by RealLucy**
I backport this for parity with 11.0.21-oracle. 
Whitebox obviously differs in location between 11 and head. Needed adaptations.
In all affected files, some chunks did not apply. Manual resolution was straightforward.

GHA and SAP-internal tests succeeded.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300659](https://bugs.openjdk.org/browse/JDK-8300659): Refactor TestMemoryAwareness to use WhiteBox api for host values (**Bug** - P3)

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/2031/head:pull/2031` \
`$ git checkout pull/2031`

Update a local copy of the PR: \
`$ git checkout pull/2031` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/2031/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2031`

View PR using the GUI difftool: \
`$ git pr show -t 2031`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/2031.diff">https://git.openjdk.org/jdk11u-dev/pull/2031.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/2031#issuecomment-1622065753)

### CODE REVIEW

File: test/hotspot/jtreg/containers/docker/TestMemoryAwareness.java [hunk 1]
```diff
@@ -35,23 +35,26 @@
  *          jdk.jartool/sun.tools.jar
  * @build AttemptOOM sun.hotspot.WhiteBox PrintContainerInfo CheckOperatingSystemMXBean
  * @run driver ClassFileInstaller -jar whitebox.jar sun.hotspot.WhiteBox sun.hotspot.WhiteBox$WhiteBoxPermission
- * @run driver TestMemoryAwareness
+ * @run main/othervm -Xbootclasspath/a:whitebox.jar -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI TestMemoryAwareness
  */
 import jdk.test.lib.containers.docker.Common;
 import jdk.test.lib.containers.docker.DockerRunOptions;
 import jdk.test.lib.containers.docker.DockerTestUtils;
+import jdk.test.lib.sun.hotspot.WhiteBox;
```
Comment (jerboaa) [inline test/hotspot/jtreg/containers/docker/TestMemoryAwareness.java:43]: Should be:

```
import sun.hotspot.WhiteBox;
```

**Review by jerboaa — APPROVED** [review]
The import mistake makes the test not compile. Looks good otherwise.

File: test/hotspot/jtreg/containers/docker/TestMemoryAwareness.java (hunk 1 above)
Comment (RealLucy) [reply test/hotspot/jtreg/containers/docker/TestMemoryAwareness.java]: Thank you for approving this PR, @jerboaa 
And thank you for pointing me to the error. I was in a hurry to make the SAP nightmare deadline. 
Learning? If you can't make it with diligence, don't do it. As always. 
import stmt is fixed.

**Review by jerboaa — APPROVED** [review]

**Pull request closed without merging**